lib/papertrail/search_query.rb

```diff
@@ -1,21 +1,48 @@
 require 'papertrail/search_result'
 
+# SearchQuery manages pagination.
+# Once initialized, call `results_page` for a page of results
+# call it again to get the next results, and again, and again
 module Papertrail
   class SearchQuery
+    def self.api_url
+      '/api/v1/events/search.json'
+    end
+
+    def self.initial_search_limit
+      100
+    end
+
+    def self.subsequent_search_limit
+      1000
+    end
+
     def initialize(connection, query = nil, options = {})
       @connection = connection
       @query      = query
       @options    = options
     end
 
-    def search
+    attr_accessor :max_id, :subsequent_request
+
+    def results_page
```

Comment:
I considered that - but it's also the thing you query for the *first* page of results.